client/components/jetpack/backup-schedule-setting/hooks.ts

```diff
@@ -0,0 +1,62 @@
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import useScheduledTimeQuery from 'calypso/data/jetpack-backup/use-scheduled-time-query';
+import { applySiteOffset } from 'calypso/lib/site/timezone';
+import { useSelector } from 'calypso/state';
+import getSiteGmtOffset from 'calypso/state/selectors/get-site-gmt-offset';
+import getSiteTimezoneValue from 'calypso/state/selectors/get-site-timezone-value';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { convertHourToRange } from './utils';
+
+const useNextBackupSchedule = () => {
+	const moment = useLocalizedMoment();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const timezone = useSelector( ( state ) => getSiteTimezoneValue( state, siteId ) );
+	const gmtOffset = useSelector( ( state ) => getSiteGmtOffset( state, siteId ) );
+
+	const { data, isSuccess } = useScheduledTimeQuery( siteId );
+
+	const getNextBackupDate = () => {
+		if ( ! data || data.scheduledHour === null ) {
+			return null;
+		}
+
+		const currentTime = moment();
+		const backupTimeUtc = moment.utc().startOf( 'day' ).hour( data.scheduledHour );
+
+		let nextBackupDate = backupTimeUtc;
+
+		// Apply site offset if available
+		if ( timezone && gmtOffset ) {
+			nextBackupDate = applySiteOffset( backupTimeUtc, { timezone, gmtOffset } );
+		} else {
+			nextBackupDate = backupTimeUtc.local();
+		}
+
+		// Check if the next backup is for tomorrow or today
+		if ( nextBackupDate.isBefore( currentTime ) ) {
```

Comment:
What happens when we are in the exact range of hours of backup execution?

I think that it will show the next backup date will be the next day, but actually, the backup may or not have been executed (since the backup can be triggered at any time in that range or hours). Maybe this case needs some specific message?

Reply:
Nice catch!

> What happens when we are in the exact range of hours of backup execution?

It is saying that the next backup will run the next day.
As discussed on Slack, I just [pushed a change](https://github.com/Automattic/wp-calypso/pull/95602/commits/534745a591cf163423241dbf773529084c559eb8) to ensure the next backup is not moved to the next day within the backup window.